monkey/agent_plugins/exploiters/snmp/src/snmp_exploiter.py

```diff
@@ -0,0 +1,124 @@
+from typing import Callable, Iterable, Sequence
+
+from common.types import AgentID, Event
+from infection_monkey.exploit import IAgentOTPProvider
+from infection_monkey.exploit.tools import HTTPBytesServer
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.utils.threading import interruptible_iter
+
+from .snmp_command_builder import build_snmp_command
+
+logger = getLogger(__name__)
+
+AgentBinaryServerFactory = Callable[[TargetHost], HTTPBytesServer]
+CommunityStringGenerator = Callable[[], Iterable[str]]
+
+
+class StubSNMPOptions:
+    def __init__(self, *args, **kwargs):
+        pass
+
+
+class StubSNMPExploitClient:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def exploit_host(self, *args, **kwargs) -> ExploiterResultData:
+        return ExploiterResultData()
+
+
+class SNMPExploiter:
+    def __init__(
+        self,
+        agent_id: AgentID,
+        exploit_client: StubSNMPExploitClient,
+        agent_binary_server_factory: AgentBinaryServerFactory,
+        generate_community_strings: CommunityStringGenerator,
+        otp_provider: IAgentOTPProvider,
+    ):
+        self._agent_id = agent_id
+        self._exploit_client = exploit_client
+        self._agent_binary_server_factory = agent_binary_server_factory
+        self._generate_community_strings = generate_community_strings
+        self._otp_provider = otp_provider
+
+    def exploit_host(
+        self,
+        host: TargetHost,
+        servers: Sequence[str],
+        current_depth: int,
+        options: StubSNMPOptions,
+        interrupt: Event,
+    ) -> ExploiterResultData:
+        try:
+            logger.debug("Starting the agent binary server")
+            agent_binary_http_server = self._agent_binary_server_factory(host)
+        except Exception as err:
+            msg = (
+                "An unexpected exception occurred while attempting to start the agent binary HTTP "
+                f"server: {err}"
+            )
+            logger.exception(msg)
+            return ExploiterResultData(error_message=msg)
+
+        command = build_snmp_command(
+            self._agent_id,
+            host,
+            servers,
+            current_depth,
+            agent_binary_http_server.download_url,
+            self._otp_provider.get_otp(),
+        )
+
+        try:
+            exploit_result = self._exploit_community_strings(
+                host,
+                options,
+                command,
+                agent_binary_http_server.bytes_downloaded,
+                interrupt,
+            )
+            return exploit_result
+        except Exception as err:
+            msg = f"An unexpected exception occurred while exploiting host {host} with SNMP: {err}"
+            logger.exception(msg)
+            return ExploiterResultData(error_message=msg)
+        finally:
+            _stop_agent_binary_http_server(agent_binary_http_server)
+
+    def _exploit_community_strings(
```

Comment:
This name is maybe misleading. We're not exploiting community strings. Maybe "_exploit_host_with_brute_force", or "_brute_force_exploit_host()" or "_try_exploit_with_all_community_strings". Or something else that's more descriptive of what this function is doing.